src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -89,34 +29,115 @@ const propTypes = {
 
     // Controls the visibility of this component.
     isVisible: PropTypes.bool,
+
+    // The session of the logged in person
+    session: PropTypes.shape({
+        // Email of the logged in person
+        email: PropTypes.string,
+    }),
 };
 
 const defaultProps = {
     reportAction: {},
     isMini: false,
     isVisible: false,
+    session: {},
 };
 
-const ReportActionContextMenu = (props) => {
-    const wrapperStyle = getReportActionContextMenuStyles(props.isMini);
-    return props.isVisible && (
-        <View style={wrapperStyle}>
-            {CONTEXT_ACTIONS.map(contextAction => contextAction.shouldShow && (
-                <ReportActionContextMenuItem
-                    icon={contextAction.icon}
-                    text={contextAction.text}
-                    successIcon={contextAction.successIcon}
-                    successText={contextAction.successText}
-                    isMini={props.isMini}
-                    onPress={() => contextAction.onPress(props.reportAction)}
-                    key={contextAction.text}
-                />
-            ))}
-        </View>
-    );
-};
+class ReportActionContextMenu extends React.Component {
+    constructor(props) {
+        super(props);
+
+        /**
+         * A list of all the context actions in this menu.
+         */
+        this.contextActions = [
+            // Copy to clipboard
+            {
+                text: 'Copy to Clipboard',
+                icon: ClipboardIcon,
+                successText: 'Copied!',
+                successIcon: Checkmark,
+                shouldShow: true,
+
+                // If return value is true, we switch the `text` and `icon` on
+                // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
+                // the `text` and `icon`
+                onPress: () => {
+                    const message = _.last(lodashGet(this.props.reportAction, 'message', null));
+                    const html = lodashGet(message, 'html', '');
+                    const text = lodashGet(message, 'text', '');
+                    const isAttachment = _.has(this.props.reportAction, 'isAttachment')
+                        ? this.props.reportAction.isAttachment
+                        : isReportMessageAttachment(text);
+                    if (!isAttachment) {
+                        Clipboard.setString(text);
+                    } else {
+                        Clipboard.setString(html);
+                    }
```

Comment:
I think this should be moved to a class method so that the constructor doesn't become bloated with a bunch of callback logic.

Reply:
> I think this should be moved to a class method so that the constructor doesn't become bloated with a bunch of callback logic.

Good point, but maybe an alternate solution to populating the context actions in the component state (which I think could delay its rendering, right?) would be to just declare plain functions above the component like:

```javascript
function copyToClipboard(reportAction) {
    const message = _.last(lodashGet(this.props.reportAction, 'message', null));
    const html = lodashGet(message, 'html', '');
    const text = lodashGet(message, 'text', '');
    ...
    ...
}

constructor(props) {
    super(props);
    this.contextActions = [
        {
                text: 'Copy to Clipboard',
                icon: ClipboardIcon,
                successText: 'Copied!',
                successIcon: Checkmark,
                shouldShow: true,
                onPress: () => copyToClipboard(props.reportAction),
           }
    ];
}
```

The potential downside to this alternate solution would be that the callbacks would be "pure functions" (i.e: wouldn't be bound to the component). But since the component doesn't need to manage state and all the callbacks are using Onyx actions, that might be fine.

Reply:
Yes, I think making them pure functions is fine as well (and maybe `this.props` is passed to every callback).

> which I think could delay its rendering, right?

I think this delay is so negligible that it's not worth considering, especially with an early return in the render method.

I like the idea of using pure functions better though and keeping these defined outside of the component.